gapir/api.go

```diff
@@ -51,24 +51,26 @@ type (
 // from a connected GAPIR device.
 type ReplayResponseHandler interface {
 	// HandlePayloadRequest handles the given payload request message.
-	HandlePayloadRequest(context.Context, string, Connection) error
+	HandlePayloadRequest(context.Context, string) error
 	// HandleResourceRequest handles the given resource request message.
-	HandleResourceRequest(context.Context, *ResourceRequest, Connection) error
+	HandleResourceRequest(context.Context, *ResourceRequest) error
 	// HandleCrashDump handles the given crash dump message.
-	HandleCrashDump(context.Context, *CrashDump, Connection) error
+	HandleCrashDump(context.Context, *CrashDump) error
 	// HandlePostData handles the given post data message.
-	HandlePostData(context.Context, *PostData, Connection) error
+	HandlePostData(context.Context, *PostData) error
 	// HandleNotification handles the given notification message.
-	HandleNotification(context.Context, *Notification, Connection) error
+	HandleNotification(context.Context, *Notification) error
 	// HandleFinished handles the replay complete
-	HandleFinished(context.Context, error, Connection) error
+	HandleFinished(context.Context, error) error
 	// HandleFenceReadyRequest handles the profiler ready message.
-	HandleFenceReadyRequest(context.Context, *FenceReadyRequest, Connection) error
+	HandleFenceReadyRequest(context.Context, *FenceReadyRequest) error
 }
 
 // Connection represents a connection between GAPIS and GAPIR. It wraps the
 // internal gRPC connections and holds authentication token. A new Connection
 // should be created only by client.Client.
+
+// TODO: We should update this comment.
```

Comment:
Given the current lack of design, the original comment does not make much sense, and brings more confusion than anything else. I would be keen to just remove it, rather than commenting on top of it.

Reply:
I am actually waiting suggestion for it. We should update before the merge.